### PR TITLE
support for node 12

### PR DESCRIPTION
**Pull request by WeiAnAn**
Fix deprecated V8 api for supporting node 12.

### CODE REVIEW

**Comment by WeiAnAn** [comment]
I'm not sure if using new API will cause error in older node version or not.
I will do some test.

**Comment by Rush** [comment]
You should probably upgrade Nan and try with it.

**Comment by WeiAnAn** [comment]
I have tested node version v0.12, v4, v6, v8, v10 and v12.
I think it work great after update.

File: authenticate_pam.cc [hunk 1]
```diff
@@ -104,7 +104,8 @@ void after_doing_auth(uv_work_t* req, int status) {
 		args[0] = Nan::New<String>(m->errorString.c_str()).ToLocalChecked();
 	}
 
-  Nan::MakeCallback(Nan::GetCurrentContext()->Global(), Nan::New(m->callback), 1, args);
+	Nan::AsyncResource *asyncResource = new Nan::AsyncResource("callback");
```
Comment (Rush) [inline authenticate_pam.cc]: At which point does this object get deleted? I am worried of creating potential memory leaks.

**Comment by artynet** [comment]
@Rush take a look at my compatiblity patch :`

https://github.com/Rush/node-authenticate-pam/pull/30

I think it can be of help for releasing a new tag. I made the control statements over the **V8_MAJOR_VERSION** release.

**Comment by Rush** [comment]
Hmm, I see conflicts now after merging the other compatibility branch.

**Pull request closed without merging**